### ISSUE TITLE
Support copy/paste for files/directories

### ISSUE THREAD

**Issue by TahaTesser** [invertase/zapp.run#51]
Please add the ability to copy/paste files/directories in the side panel. So files/directories can be duplicated too.

![Screenshot 2022-07-19 at 11 14 30](https://user-images.githubusercontent.com/48603081/179702064-2ed82b18-fb83-4da0-9846-bc65caf5fd53.png)

- Copy and paste in the same folder should create a copy with an updated filename.
  - `home.dart`
  _copy and pasted_ 
   - `home_copy.dart`
- Ability to copy and paste files in different folders.
- Can use shortcuts to copy/paste files.

Related to:
https://github.com/invertase/zapp.run/issues/10 
https://github.com/invertase/zapp.run/issues/48

**Comment by Ehesp** [comment]
Since copy/paste only works in the context of Zapp, it might be a better DX to just have a `Duplicate` button which does a copy then paste?

**Comment by Salakar** [comment]
> Since copy/paste only works in the context of Zapp, it might be a better DX to just have a `Duplicate` button which does a copy then paste?

This seems like a non-standard devex, most developers would be familiar with a copy & paste flow

**Comment by Ehesp** [comment]
Does this work instead?

- Copy: does 2 things, copies file contents to clipboard + stores the file in state.
- Paste: Duplicates the copied file

**Comment by Salakar** [comment]
> Does this work instead?
> 
> * Copy: does 2 things, copies file contents to clipboard + stores the file in state.
> * Paste: Duplicates the copied file

I don't think we need to do anything clipboard related, based on vscode.dev copy and paste only affects the file tree, e.g. https://vscode.dev/github/invertase/denque ;

e.g. Copy stores some `explorerCopiedFileOrDirectory` state, paste just reads that state and does whatever
